Arduino/Drv/PwmDriver/PwmDriver.cpp

```diff
@@ -0,0 +1,56 @@
+// ======================================================================
+// \title  PwmDriver.cpp
+// \author ethan
+// \brief  cpp file for PwmDriver component implementation class
+// ======================================================================
+
+
+#include <Arduino/Drv/PwmDriver/PwmDriver.hpp>
+#include <FpConfig.hpp>
+#include <FprimeArduino.hpp>
+
+namespace Arduino {
+
+  // ----------------------------------------------------------------------
+  // Construction, initialization, and destruction
+  // ----------------------------------------------------------------------
+
+  PwmDriver ::
+    PwmDriver(
+        const char *const compName
+    ) : PwmDriverComponentBase(compName), m_pin(-1)
+  {
+
+  }
+
+  PwmDriver ::
+    ~PwmDriver()
+  {
+
+  }
+
+  void PwmDriver::open(NATIVE_INT_TYPE gpio)
+  {
+    this->m_pin = gpio;
+    pinMode(this->m_pin, Arduino::DEF_OUTPUT);
+  }
+
+  // ----------------------------------------------------------------------
+  // Handler implementations for user-defined typed input ports
+  // ----------------------------------------------------------------------
+
+  Fw::Success PwmDriver ::
+    setDutyCycle_handler(
+        const NATIVE_INT_TYPE portNum,
+        U8 dutyCycle
+    )
+  {
+    if (dutyCycle > 100) {
+        dutyCycle = 100;
+    }
+
+    analogWrite(this->m_pin, 255 * (static_cast<F32>(dutyCycle) / 100));
```

Comment:
I prefer to writ this statement as:

```
(255 * dutyCycle)/100
```

In this way you avoid floating-point math while achieving the same result.